packages/fixie/src/agent.ts

```diff
@@ -113,68 +113,42 @@ export class FixieAgent extends FixieAgentBase {
   }
 
   /**
-   * Create a new agent revision.
+   * Create a new managed agent revision from a code package.
    *
+   * @param tarball The path to the tarball containing the agent code.
    * @param defaultRuntimeParameters The default runtime parameters for the agent.
-   * @param externalUrl The URL of the agent's external deployment, if any.
    * @param runtimeParametersSchema The JSON-encoded schema for the agent's runtime parameters.
-   *   May only be specified if `externalUrl` is also specified.
    * @param isCurrent Whether this revision should be the current revision.
-   * @param tarball The path to the tarball containing the agent code.
    * @param environmentVariables The environment variables to set for the agent.
-   *   May only be specified if `tarball` is also specified.
    *
    * @returns The created agent revision.
    */
-  public async createRevision({
+  public async createManagedRevision({
```

Comment:
How does `fixie serve` work without the `externalUrl` support?

Reply:
It doesn't use this code path. It uses the base class `createRevision` instead. See line 431.

Reply:
I see -- I don't really understand why the code is split across the two packages the way it is, but presumably you have a reason :)